src/fu-main.c

```diff
@@ -2160,15 +2200,41 @@ main(int argc, char *argv[])
 	}
 
 	/* own the object */
-	priv->owner_id = g_bus_own_name(G_BUS_TYPE_SYSTEM,
-					FWUPD_DBUS_SERVICE,
-					G_BUS_NAME_OWNER_FLAGS_ALLOW_REPLACEMENT |
-					    G_BUS_NAME_OWNER_FLAGS_REPLACE,
-					fu_main_on_bus_acquired_cb,
-					fu_main_on_name_acquired_cb,
-					fu_main_on_name_lost_cb,
-					priv,
-					NULL);
+	if (socket_filename != NULL) {
+		g_autofree gchar *address = g_strdup_printf("unix:path=%s", socket_filename);
+		g_autofree gchar *guid = g_dbus_generate_guid();
+		g_autoptr(GDBusServer) server = NULL;
+
+		/* this must be owned by root */
+		g_unlink(socket_filename);
```

Comment:
I would think you want some safety checks to make sure that the file being unlinked is a socket or so no?

Reply:
I'm wondering if we want to do the unlink at all; it seems to me that writing a systemd service that manages the socket outside of the daemon seems to be the 2022 way to do it.

Reply:
Thinking about flexibility how about check if compiled with systemd and do that, and if not do the unlink?